db/blob/blob_source.cc

```diff
@@ -204,22 +204,70 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
 
 void BlobSource::MultiGetBlob(
     const ReadOptions& read_options,
-    const autovector<std::reference_wrapper<const Slice>>& user_keys,
-    uint64_t file_number, uint64_t file_size,
-    const autovector<uint64_t>& offsets,
-    const autovector<uint64_t>& value_sizes, autovector<Status*>& statuses,
-    autovector<PinnableSlice*>& blobs, uint64_t* bytes_read) {
-  size_t num_blobs = user_keys.size();
+    std::map<uint64_t, autovector<BlobReadRequest>>& blob_reqs,
+    uint64_t* bytes_read) {
+  assert(blob_reqs.size() > 0);
+
+  Status s;
+
+  for (auto& batch_rqs : blob_reqs) {
+    const uint64_t file_number = batch_rqs.first;
+    CacheHandleGuard<BlobFileReader> blob_file_reader;
+    s = GetBlobFileReader(file_number, &blob_file_reader);
+    assert(!s.ok() || blob_file_reader.GetValue());
+
+    auto& blob_reqs_in_file = batch_rqs.second;
+    if (!s.ok()) {
+      for (const auto& blob_req : blob_reqs_in_file) {
+        *blob_req.status = s;
+      }
+      continue;
+    }
+
+    assert(blob_file_reader.GetValue());
+    const uint64_t file_size = blob_file_reader.GetValue()->GetFileSize();
```

Comment:
This works but we could also use the file size from the MANIFEST to be in sync with `GetBlob`. To do that, this method could take an `autovector<std::tuple<uint64_t /* file_number */, uint64_t /* file_size */, autovector<BlobReadRequest>>`

Reply:
Make sense. Thanks a lot!

Reply:
Done